fabtests/multinode/src/core.c

```diff
@@ -123,20 +156,67 @@ static int multinode_setup_fabric(int argc, char **argv)
 		goto err;
 	}
 
-	ret = fi_av_insert(av, pm_job.names, pm_job.num_ranks,
-			   pm_job.fi_addrs, 0, NULL);
-	if (ret != pm_job.num_ranks) {
-		FT_ERR("unable to insert all addresses into AV table\n");
-		ret = -1;
+	for (i = 0; i < pm_job.num_ranks; i++) {
+		ret = fi_av_insert(av, (char*)pm_job.names + i * pm_job.name_len, 1,
+			   &pm_job.fi_addrs[i], 0, NULL);
+		if (ret != 1) {
+			FT_ERR("unable to insert all addresses into AV table\n");
+			ret = -1;
+			goto err;
+		}
+	}
+
+	pm_job.multi_iovs = malloc(sizeof(struct fi_rma_iov) * pm_job.num_ranks);
```

Comment:
Use sizeof(*variable) when possible.  Avoid sizeof(struct blah).  The former is easier to maintain.